lib/src/Uncertainty/Distribution/UniformOverMesh.cxx

```diff
@@ -0,0 +1,214 @@
+{
+  OSS oss(false);
+  oss << getClassName() << "(mesh = " << mesh_ << ")";
+  return oss;
+}
+
+/* Virtual constructor */
+UniformOverMesh * UniformOverMesh::clone() const
+{
+  return new UniformOverMesh(*this);
+}
+
+/* Compute the numerical range of the distribution given the parameters values */
+void UniformOverMesh::computeRange()
+{
+  setRange(Interval(mesh_.getLowerBound(), mesh_.getUpperBound()));
+}
+
+
+/* Get one realization of the distribution */
+Point UniformOverMesh::getRealization() const
+{
+  const UnsignedInteger index = simplexSelection_.getRealization()[0];
+  const Indices simplex(mesh_.getSimplex(index));
+  const Sample vertices(mesh_.getVertices().select(simplex));
+  return DistFunc::rUniformSimplex(vertices);
+}
+
+
+/* Get the PDF of the distribution, i.e. P(point < X < point+dx) = PDF(point)dx + o(dx) */
+Scalar UniformOverMesh::computePDF(const Point & point) const
+{
+  if (point.getDimension() != getDimension()) throw InvalidArgumentException(HERE) << "Error: the given point must have dimension=" << getDimension() << ", here dimension=" << point.getDimension();
+  Scalar pdf = 0.0;
+  if (meshDomain_.contains(point))
+    pdf = 1.0 / meshVolume_;
+  return pdf;
+}
+
+
+/* Get the CDF of the distribution, i.e. P(X <= point) = CDF(point) */
+Scalar UniformOverMesh::computeCDF(const Point & point) const
+{
+  if (point.getDimension() != getDimension()) throw InvalidArgumentException(HERE) << "Error: the given point must have dimension=" << getDimension() << ", here dimension=" << point.getDimension();
+  Scalar cdf = 0.0;
+  // Waiting for a better implementation
+  const Interval i(Point(getDimension(), -SpecFunc::MaxScalar), point);
+  const Interval intersection(i.intersect(getRange()));
+  if (intersection == getRange()) cdf = 1.0;
+  else if (!intersection.isEmpty())
+    cdf = integrationAlgorithm_.integrate(PDFWrapper(this), intersection)[0];
+  return cdf;
+}
+
+/* Compute the mean of the distribution */
+void UniformOverMesh::computeMean() const
+{
+  const UnsignedInteger dimension = getDimension();
+  mean_ = Point(dimension);
+  for (UnsignedInteger i = 0; i < simplicesVolumes_.getSize(); ++i)
+  {
+    const Indices simplex(mesh_.getSimplex(i));
+    Point center(dimension);
+    for (UnsignedInteger j = 0; j <= dimension; ++j)
+      center += mesh_.getVertex(simplex[j]);
+    mean_ += (simplicesVolumes_[i] / (meshVolume_ * (1.0 + dimension))) * center;
+  }
+  isAlreadyComputedMean_ = true;
+}
+
+/* Mesh accessor */
+void UniformOverMesh::setMesh(const Mesh & mesh)
+{
+  setDimension(mesh.getDimension());
+  mesh_ = mesh;
+  meshDomain_ = MeshDomain(mesh);
+  simplicesVolumes_ = mesh.computeSimplicesVolume();
+  meshVolume_ = std::accumulate(simplicesVolumes_.begin(), simplicesVolumes_.end(), 0.0);
+  simplexSelection_ = UserDefined(RegularGrid(0.0, 1.0, simplicesVolumes_.getSize()).getVertices(), simplicesVolumes_);
+  uniformSimplex_ = Dirichlet(Point(getDimension() + 1, 1.0));
```

Comment:
I can vaguely see why that attribute is there, but it is not used by the algorithm within the class and there is no accessor: is is a temporary debug code ? shall it be removed?

Reply:
Good catch! It was used to sample the distribution, but the service has been optimized and moved to the DistFunc class.